### DISCUSSION TITLE
Eventlistener i.e. OnSessionBegin

### DISCUSSION THREAD

**Discussion by thefirstfloor** [immersivecognition/unity-experiment-framework#73] [Q&A] [answered]
Linking my public methods to the event slots in the UXF inspector tab 'events' works fine, but I'd like to define the event listener in code so I won't need the inspector for (all) of the methods I want to be called.
I see it's a custom class sessionevent, which is in turn an unityevent, but am uncertain on how to approach this.

**Comment by jackbrookes** [comment, marked as answer]
it should just work the same way as a unity event:

https://docs.unity3d.com/ScriptReference/Events.UnityEvent.AddListener.html

However they need to take the Session as an argument e.g.

```cs

void Start()
{
    session.onSessionBegin.AddListener(DoStuff);
}

void DoStuff(Session mySession)
{
    Debug.Log("Hello");
}

```


or even

```cs
void Start()
{
    session.onSessionBegin.AddListener((s) => Debug.Log("Hello"));
}
```

Note though, these listeners will be added, but will not show up the UI. They will be removed when you exit play mode.

**Reply by thefirstfloor** [reply]
Thanks! It must have been the Monday as this is pretty obvious.

Also possible:
`Session.instance.onSessionBegin.AddListener(DoStuff);`
